### PR TITLE
Fix CI warnings and CodeCov

### PR DESCRIPTION
**Pull request by shralex**
Resolves observed DeprecationWarnings and UserWarnings across unit and integration tests. Fixed the Codecov CI pipeline.

Key Changes:

- Flax NNX Modernization: Replaced all deprecated .value accesses on nnx.Variable instances with the indexing syntax [...] or .get_value() throughout the maxtext models and test suites.
- Sharding Annotations: Swapped out the deprecated sharding keyword parameter for the newly supported out_sharding explicitly in nnx.Cache and nnx.Param instantiations to align with standard JAX naming conventions.
- Pytree State Refactor: Updated internal state tracking logic from _object__state to the modern _pytree__state.
- JAX UserWarnings: Mitigated JAX integer truncation warnings in the DeepSeek engram.py layer by enforcing explicit jnp.int32 definitions for token and hash index allocations.
- Repaired the CodeCov integration that recently ceased running to restore code coverage tracking.

# Tests

Tested on TPU VM

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by igorts-git — APPROVED** [review]

File: src/maxtext/configs/types.py [hunk 1]
```diff
@@ -847,7 +847,7 @@ class HardwareAndMesh(BaseModel):
       "all_gather",
       description="Strategy for context parallelism ('all_gather' or 'ring').",
   )
-  context_parallel_reorder_strategy: ReorderStrategy = Field(
+  context_parallel_reorder_strategy: str = Field(
```
Comment (ecnal-cienet) [inline src/maxtext/configs/types.py]: LGTM

Weakening the type from `ReorderStrategy` to `str` silently breaks the equality checks downstream. With `ConfigDict(...)` not setting `validate_default=True`, Pydantic only coerces strings → enum on explicit user input — so before this PR, `Config(f="striped").f` was `ReorderStrategy.STRIPED` and comparisons worked; after, it stays as the raw string and equality is False.

I think this minimun fix might work:
```python
context_parallel_reorder_strategy: ReorderStrategy = Field(
    ReorderStrategy.AUTO, description="..."
)
```

**Review by richjames0 — APPROVED** [review]
lgtm

File: src/maxtext/layers/engram.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ class CompressedTokenizer:
   def __init__(self, tokenizer: HFTokenizer):
     normalizer = self._build_normalizer()
     self.lookup_table_np, self.num_new_token = self._build_lookup_table(tokenizer, normalizer)
-    self.lookup_table = jnp.array(self.lookup_table_np, dtype=jnp.int64)
```
Comment (RissyRan) [inline src/maxtext/layers/engram.py:56]: If I recall correctly for previous runs, the reference [test](https://github.com/AI-Hypercomputer/maxtext/blob/1adb45ce021c39103c0266854189f47f79c88386/tests/unit/engram_vs_reference_test.py#L173) will fail with different type. 

I am fine to update int64 to int32, as currently Engram is not used in OSS models. For future debugging purpose, could we create a bug and track this int64 changes in lines? So we could convert it back in the future

**Review by RissyRan — APPROVED** [review]
Thanks Alex for the fix! Just one comment about int64 change.

**Review by bvandermoon — APPROVED** [review]
